Cart

desafree opened the pull request:
Implement cart functionalities. It is now possible to add items from the cart popup or in the product detail page

desafree committed:
resabe: resolve conflict in shared index

desafree committed:
refactor: improve readability and modularity

-replace button element with button component in navbardesktop
-adjust style for buttonproduct
-replace div element with box component in cartitem component
-replace findindex with find in cart slice
-remove unecessary variable in useselectproductfromcart
-improve name for selectproductbyid selector

desafree committed:
rebase: resolve conflict in shared index

desafree committed:
rebase: resolve conflict in button component

desafree committed:
feat: add item from product page functionality

desafree committed:
feat: disable button when no items in cart

desafree committed:
refactor: improve readability and modularity

-handle event propagation in usetoggle
-improve style of buttonitem
-add translation for remove all
-remove unnecessary import in cart slice

desafree committed:
refactor: change state update in useToggle

desafree committed:
refactor: create hook with selectproductfromcart selector

desafree committed:
fix: resolve event propagation bug on popup

desafree committed:
refactor: replace useselectproductfromcart hook with selectproductsfromcart selector

rams23 commented on src/_shared/utils/defineStyleForButton.ts:
```diff
@@ -1,6 +1,14 @@
+      minWidth: "0",
+      padding: "0",
+    };
+  }
```
this can be done inside te theme

rams23 commented on src/cart/components/CartInfo/CartInfo.styled.tsx:
```diff
@@ -0,0 +1,8 @@
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 32px;
+`;
```
This can be done with stack from mui

rams23 commented on src/cart/components/CartPopup/CartPopup.tsx:
```diff
@@ -0,0 +1,25 @@
+}
+
+export const CartPopup: FC<Props> = ({ toggle }) => {
+  const items = useSelector(selectProductsFromCart);
```
redux in components, better a custom hook

rams23 commented on src/cart/components/Items/Items.styled.tsx:
```diff
@@ -0,0 +1,8 @@
+import styled from "styled-components";
+
+export const ItemsStyled = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 16px;
+  margin-bottom: 32px;
+`;
```
also this is a stack I think

rams23 commented on src/cart/slice/slice.ts:
```diff
@@ -1,8 +1,12 @@
+import { selectIProductsSlice } from "../../products";
 
-const initialState: CartProduct[] = [{ id: 1, qty: 10 }];
+const initialState: CartProduct[] = [
+  { id: 1, qty: 10 },
+  { id: 2, qty: 10 },
+];
```
considering that we will access it by id, we can improve the state structure using a map instead of an array

rams23 commented on src/products/components/CartActions/CartActions.tsx:
```diff
@@ -1,15 +1,42 @@
+        handleClickPlus={handleClickPlus}
+      />
+      <Button
+        variant="orange"
+        onClick={() => {
```
conceptually orange is better to be a color instead of a variant, if we want to strict consider the design system

rams23 commented on src/cart/slice/slice.ts:
```diff
@@ -34,5 +51,24 @@ const slice = createSlice({
 export const selectCartSlice = (state: RootState) => state.cart;
 
+export const selectProductsFromCart = createSelector(
+  [selectCartSlice, selectIProductsSlice],
```
Why the "I" in the selector name, moreover a slice should not be aware of another slice structure, so we should use a more specific selector, not the slice but the products list

desafree committed:
reafactor: define button variant in theme

desafree committed:
refactor: define link variants in theme

desafree committed:
refactor: replace flex style with stack component

desafree committed:
refactor: improve modularity and clearness

-create a custom hook to remove redux in components
-improved state structure using createentityadapter
